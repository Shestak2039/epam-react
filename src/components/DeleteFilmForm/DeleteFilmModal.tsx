import React from 'react';

import './delete-film-form.scss';
import Button from '../common/Button/Button';

const DeleteFilmModal: React.FunctionComponent = () => (
    <div className="delete-film-form">
        <h2 className="header-title">DELETE MOVIE</h2>
        <h4 className="confirm-text">Are you sure you want to delete this movie?</h4>
        <div className="buttons-area">
            <Button title={'CONFIRM'} primary={true} />
        </div>
    </div>
);

export default DeleteFilmModal;
