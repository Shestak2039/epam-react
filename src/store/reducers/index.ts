import { combineReducers } from 'redux';

import filmsReducer from './films-reducer';

export default combineReducers({
    filmsState: filmsReducer
});
