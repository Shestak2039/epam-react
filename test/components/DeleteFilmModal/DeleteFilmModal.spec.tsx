import React from 'react';
import { render, waitFor } from '@testing-library/react';
import DeleteFilmModal from '../../../src/components/DeleteFilmForm/DeleteFilmModal';
import { Provider } from 'react-redux';
import store from '../../../src/store/store';
import userEvent from '@testing-library/user-event/dist';


describe('DeleteFilmModal', () => {
    test('renders DeleteFilmModal snapshot', async () => {
        const closeModal = jest.fn();
        const deleteFilm = jest.fn();
        // @ts-ignore
        const { asFragment, getByTestId } = render(<Provider store={store}><DeleteFilmModal filmId={'1'} closeModal={closeModal} deleteFilm={deleteFilm}/></Provider>);

        userEvent.click(getByTestId('CONFIRM'));

        expect(asFragment()).toMatchSnapshot();
        
        waitFor(() => {
            expect(closeModal).toHaveBeenCalled();
            expect(deleteFilm).toHaveBeenCalled();
        })
    });
});
