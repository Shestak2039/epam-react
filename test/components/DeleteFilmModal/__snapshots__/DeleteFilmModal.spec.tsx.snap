// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFilmModal renders DeleteFilmModal snapshot 1`] = `
<DocumentFragment>
  <div
    class="delete-film-form"
  >
    <h2
      class="header-title"
    >
      DELETE MOVIE
    </h2>
    <h4
      class="confirm-text"
    >
      Are you sure you want to delete this movie?
    </h4>
    <div
      class="buttons-area"
    >
      <button
        class="button primary"
        data-testid="CONFIRM"
      >
        CONFIRM
      </button>
    </div>
  </div>
</DocumentFragment>
`;
